### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by RavisBlue**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,3 +3,45 @@
...
+    - Users 5-8 demand Ultra Reliable and Low Latency Communications (URLLC) traffic
+- UE mobility: static
+- Traffic: The traffic is twinned from a public dataset of LTE network from 3 BSs located in Madrid, Spain.
+    - BS1: carrier frequency at 816 MHz
+    - BS2: carrier frequency at 1835 MHz
+    - BS3: carrier frequency at 2650 MHz
+- Number of slices for the BS: 2 (slice_id 0 for eMBB UEs and slice_id 1 for URLLC UEs)
+- Slicing configurations: There are 50 PRBs in total to allocate for the two slices.
+
+|  Slice Resouce Allocation  | Slice_id 0 (eMBB: UEs 1-4) PRBs  |  Slice_id 1 (URLLC: UEs 5-8) PRBs  |
```
Comment (leonardobonati) [inline README.md]:
```suggestion
|  Slice Resouce Allocation  | Slice ID 0 (eMBB: UEs 1-4) PRBs  |  Slice ID 1 (URLLC: UEs 5-8) PRBs  |
```

File: README.md [hunk 2]
```diff
@@ -3,3 +3,45 @@
...
+|  Slice Resouce Allocation  | Slice ID 0 (eMBB: UEs 1-4) PRBs  |  Slice ID 1 (URLLC: UEs 5-8) PRBs  |
+|------------|------------|------------|
+| Slicing 1 | 9 | 41 |
+| Slicing 2 | 21 | 29 |
+| Slicing 3 | 30 | 20 |
+| Slicing 4 | 39 | 11 |
+| Slicing 5 | 50 | 0 |
+
+# Dataset stucture
+- The dataset has the following structure: 3 clusters, 5 slicings, 2 schedualing. In total 30 combinations.
```
Comment (leonardobonati) [inline README.md]: The first bullet point is not easy to understand. Can we elaborate more on this?

File: README.md [hunk 3]
```diff
@@ -3,3 +3,45 @@
...
+    - cluster/slicing/schedualing
+- Each experiment has a directory named `RESERVATION-<experiment_id>`
+- Inside each experiment directory, there is a directory named `bs` for the base station files, and there is one directory named `ue_<IMSI>` for each UE for the files related to that UE.
+- Base Station files:
+    - `mgen-script-scope.mgen`: The MGEN script that twins the real-world traffic used for generating traffic for all the UEs in the experiment. 
+    - `<IMSI>_metrics.csv`: PHY- and MAC-layer KMPs collected at the base station for each UE.
+    - `end_metrics.csv`: downlink bitrate and uplink bitrate for all the UEs in the experiment.
```
Comment (leonardobonati) [inline README.md]:
```suggestion
    - `enb_metrics.csv`: downlink and uplink bitrate of the network, aggregated for all the UEs served by the base station.
```

File: README.md [hunk 4]
```diff
@@ -3,3 +3,45 @@
...
+    - `enb.log`: log file collected at the base station.
```
Comment (leonardobonati) [inline README.md]:
```suggestion
    - `enb.log`: log file of the base station protocol stack.
```

File: README.md [hunk 5]
```diff
@@ -3,3 +3,45 @@
...
+- UE files: 
+    - `mgen.log`: MGEN logs for the UE that are from the APP layer.
+    - `mgen.csv`: Same mgen.log data converted to a csv table.
+    - `ue_metrics.csv`: PHY- and MAC-layer KPMs collected at the ue side. (?)
+    - `ue.log`: log file collected at each UE.
```
Comment (leonardobonati) [inline README.md]:
```suggestion
    - `ue.log`: log file of the UE protocol stack.
```

File: README.md [hunk 6]
```diff
@@ -3,3 +3,45 @@
...
+    - cluster/slicing/schedualing
+- Each experiment has a directory named `RESERVATION-<experiment_id>`
+- Inside each experiment directory, there is a directory named `bs` for the base station files, and there is one directory named `ue_<IMSI>` for each UE for the files related to that UE.
+- Base Station files:
+    - `mgen-script-scope.mgen`: The MGEN script that twins the real-world traffic used for generating traffic for all the UEs in the experiment. 
+    - `<IMSI>_metrics.csv`: PHY- and MAC-layer KMPs collected at the base station for each UE.
+    - `end_metrics.csv`: downlink bitrate and uplink bitrate for all the UEs in the experiment.
+    - `enb.log`: log file collected at the base station.
+- UE files: 
+    - `mgen.log`: MGEN logs for the UE that are from the APP layer.
```
Comment (leonardobonati) [inline README.md]:
```suggestion
    - `mgen.log`: Application-layer KPMs collected from MGEN.
```

File: README.md [hunk 7]
```diff
@@ -3,3 +3,45 @@
...
+    - `mgen.csv`: Same mgen.log data converted to a csv table.
```
Comment (leonardobonati) [inline README.md]: I don't think we have both, did you try to download the public dataset and check?

Comment (RavisBlue) [reply README.md]: No, we didn't. I'm changing this.

File: README.md [hunk 8]
```diff
@@ -3,3 +3,86 @@
...
+│   │   │   │   │   ├── 1010123456008_metrics.csv
+│   │   │   │   │   ├── 1010123456009_metrics.csv
+│   │   │   │   │   ├── enb.log
+│   │   │   │   │   ├── enb_metrics.csv
+│   │   │   │   │   ├── mgen-script-scope.mgn
+│   │   │   │   │   └── mgen.log
+│   │   │   │   ├── ue_001010123456002
+│   │   │   │   │   ├── mgen-script-scope.mgn
+│   │   │   │   │   ├── mgen.csv
+│   │   │   │   │   ├── mgen.log
```
Comment (leonardobonati) [inline README.md]:
```suggestion
```

File: README.md [hunk 9]
```diff
@@ -3,3 +3,86 @@
...
+```
+- Each experiment has a directory named `RESERVATION-<experiment_id>`
+- Inside each experiment directory, there is a directory named `bs` for the base station files, and there is one directory named `ue_<IMSI>` for each UE for the files related to that UE.
+- Base Station files:
+    - `mgen-script-scope.mgen`: The MGEN script that twins the real-world traffic used for generating traffic for all the UEs in the experiment. 
+    - `<IMSI>_metrics.csv`: PHY- and MAC-layer KMPs collected at the base station for each UE.
+    - `enb_metrics.csv`: downlink and uplink bitrate of the network, aggregated for all the UEs served by the base station.
+    - `enb.log`: log file of the base station protocol stack.
+- UE files: 
+    - `mgen.log`: Application-layer KPMs collected from MGEN.
```
Comment (leonardobonati) [inline README.md:86]: Say that we converted it and add the .csv one

File: README.md [hunk 10]
```diff
@@ -3,3 +3,86 @@
...
+    - `ue_metrics.csv`: PHY- and MAC-layer KPMs collected at the ue side. (?)
```
Comment (leonardobonati) [inline README.md]:
```suggestion
    - `ue_metrics.csv`: PHY- and MAC-layer KPMs collected at the ue side.
```

Also say that we converted it.

Comment (leonardobonati) [reply README.md]: We also have the mgen script at the user

File: README.md (hunk 9 above)
Comment (leonardobonati) [inline README.md]:
```suggestion
```

**Review by leonardobonati — CHANGES_REQUESTED** [review]
Thank you for putting this together. I left a few suggestions in the Readme file.

File: README.md [hunk 11]
```diff
@@ -3,3 +3,99 @@
...
+│   │   │   │   │   └── enb.log
+│   │   │   │   ├── ue_001010123456002
+│   │   │   │   │   └── ue.log
+...
+```
+
+- Each experiment has a directory named `RESERVATION-<experiment_id>`
+- Inside each experiment directory, there is a directory named `bs` for the base station files, and there is one directory named `ue_<IMSI>` for each UE for the files related to that UE.
+- Base Station files:
+    - `mgen-script-scope.mgen`: The MGEN script that twins the real-world traffic used for generating traffic for all the UEs in the experiment.
```
Comment (leonardobonati) [inline README.md]:
```suggestion
    - `mgen-script-scope.mgen`: MGEN script that twins the real-world traffic used for generating downlink traffic for the UEs in the experiment. 
```

File: README.md [hunk 12]
```diff
@@ -3,3 +3,99 @@
...
+
+- Each experiment has a directory named `RESERVATION-<experiment_id>`
+- Inside each experiment directory, there is a directory named `bs` for the base station files, and there is one directory named `ue_<IMSI>` for each UE for the files related to that UE.
+- Base Station files:
+    - `mgen-script-scope.mgen`: The MGEN script that twins the real-world traffic used for generating traffic for all the UEs in the experiment. 
+    - `<IMSI>_metrics.csv`: PHY- and MAC-layer KMPs collected at the base station for each UE.
+    - `enb_metrics.csv`: downlink and uplink bitrate of the network, aggregated for all the UEs served by the base station.
+    - `enb.log`: log file of the base station protocol stack.
+- UE files: 
+    - `mgen-script-scope.mgn`: The MGEN script that is used to indicate the listening port for each UE.
```
Comment (leonardobonati) [inline README.md]:
```suggestion
    - `mgen-script-scope.mgn`: MGEN script running at the UE, used to start the MGEN listening process.
```

File: README.md [hunk 13]
```diff
@@ -3,3 +3,99 @@
...
+    - `mgen.csv`: Application-layer KPMs collected from MGEN logs and converted to a .csv file.
```
Comment (leonardobonati) [inline README.md]:
```suggestion
    - `mgen.csv`: Application-layer KPMs collected from the MGEN logs and converted to a .csv file.
```

File: README.md [hunk 14]
```diff
@@ -3,3 +3,99 @@
...
+    - `ue_metrics.csv`: PHY- and MAC-layer KPMs collected at the ue side.
```
Comment (leonardobonati) [inline README.md]:
```suggestion
    - `ue_metrics.csv`: PHY- and MAC-layer KPMs collected at the UE side.
```

File: README.md [hunk 15]
```diff
@@ -3,3 +3,99 @@
...
+    - `ue.log`: log file of the UE protocol stack.
```
Comment (leonardobonati) [inline README.md]:
```suggestion
    - `ue.log`: log file of the UE cellular protocol stack.
```

File: README.md [hunk 16]
```diff
@@ -3,3 +3,99 @@
...
+│   │   │   │   ├── ue_001010123456002
+│   │   │   │   │   └── ue.log
+...
+```
+
+- Each experiment has a directory named `RESERVATION-<experiment_id>`
+- Inside each experiment directory, there is a directory named `bs` for the base station files, and there is one directory named `ue_<IMSI>` for each UE for the files related to that UE.
+- Base Station files:
+    - `mgen-script-scope.mgen`: The MGEN script that twins the real-world traffic used for generating traffic for all the UEs in the experiment. 
+    - `<IMSI>_metrics.csv`: PHY- and MAC-layer KMPs collected at the base station for each UE.
```
Comment (leonardobonati) [inline README.md]:
```suggestion
    - `<IMSI>_metrics.csv`: PHY- and MAC-layer KPMs collected at the base station for each UE.
```

File: README.md [hunk 17]
```diff
@@ -3,3 +3,99 @@
...
+│   │   │   ├── RESERVATION-142634
+│   │   │   │   ├── bs
+│   │   │   │   │   └── enb.log
+│   │   │   │   ├── ue_001010123456002
+│   │   │   │   │   └── ue.log
+...
+```
+
+- Each experiment has a directory named `RESERVATION-<experiment_id>`
+- Inside each experiment directory, there is a directory named `bs` for the base station files, and there is one directory named `ue_<IMSI>` for each UE for the files related to that UE.
```
Comment (leonardobonati) [inline README.md]:
```suggestion
- Inside each experiment directory, there is a directory named `bs` for the base station files, as well as directories named `ue_<IMSI>`, one for each UE, that contains the files related to the specific UE.
```

File: README.md [hunk 18]
```diff
@@ -3,3 +3,99 @@
...
+|  Slice Resouce Allocation  | Slice ID 0 (eMBB: UEs 1-4) PRBs  |  Slice ID 1 (URLLC: UEs 5-8) PRBs  |
+|------------|------------|------------|
+| Slicing 1 | 9 | 41 |
+| Slicing 2 | 21 | 29 |
+| Slicing 3 | 30 | 20 |
+| Slicing 4 | 39 | 11 |
+| Slicing 5 | 50 | 0 |
+
+# Dataset stucture
+- There are two zip files available for downloading:
```
Comment (leonardobonati) [inline README.md]:
```suggestion
- There are two compressed files available for downloading:
```

File: README.md [hunk 19]
```diff
@@ -3,3 +3,99 @@
...
+    - `open-ran-commercial-traffic-twinning-dataset-kmp` which contains the metrics and the MGEN scripts.
```
Comment (leonardobonati) [inline README.md]:
```suggestion
    - `open-ran-commercial-traffic-twinning-dataset-kmp`, which contains the metrics and the MGEN scripts.
```

File: README.md [hunk 20]
```diff
@@ -3,3 +3,99 @@
...
+    - `open-ran-commercial-traffic-twinning-dataset-log` which contains the log files.
```
Comment (leonardobonati) [inline README.md]:
```suggestion
    - `open-ran-commercial-traffic-twinning-dataset-log`, which contains the log files. 
```

File: README.md [hunk 21]
```diff
@@ -3,3 +3,99 @@
...
+| Slicing 5 | 50 | 0 |
+
+# Dataset stucture
+- There are two zip files available for downloading:
+    - `open-ran-commercial-traffic-twinning-dataset-kmp` which contains the metrics and the MGEN scripts.
+    - `open-ran-commercial-traffic-twinning-dataset-log` which contains the log files. 
+- The dataset has the following structure: cluster_<#>/slicing_<#>/schedualing_<#>
+    - 3 clusters (cluster_1, cluster_2, cluster_3)
+    - 5 slicings (slicing_1, slicing_2, slicing_3, slicing_4, slicing_5)
+    - 2 schedualings (scheduling_0, scheduling_2)
```
Comment (leonardobonati) [inline README.md]:
```suggestion
    - 2 scheduling policies (scheduling_0, scheduling_2)
```

File: README.md [hunk 22]
```diff
@@ -3,3 +3,99 @@
...
+| Slicing 4 | 39 | 11 |
+| Slicing 5 | 50 | 0 |
+
+# Dataset stucture
+- There are two zip files available for downloading:
+    - `open-ran-commercial-traffic-twinning-dataset-kmp` which contains the metrics and the MGEN scripts.
+    - `open-ran-commercial-traffic-twinning-dataset-log` which contains the log files. 
+- The dataset has the following structure: cluster_<#>/slicing_<#>/schedualing_<#>
+    - 3 clusters (cluster_1, cluster_2, cluster_3)
+    - 5 slicings (slicing_1, slicing_2, slicing_3, slicing_4, slicing_5)
```
Comment (leonardobonati) [inline README.md]:
```suggestion
    - 5 slicing configurations (slicing_1, slicing_2, slicing_3, slicing_4, slicing_5)
```

File: README.md [hunk 23]
```diff
@@ -3,3 +3,99 @@
...
+| Slicing 3 | 30 | 20 |
+| Slicing 4 | 39 | 11 |
+| Slicing 5 | 50 | 0 |
+
+# Dataset stucture
+- There are two zip files available for downloading:
+    - `open-ran-commercial-traffic-twinning-dataset-kmp` which contains the metrics and the MGEN scripts.
+    - `open-ran-commercial-traffic-twinning-dataset-log` which contains the log files. 
+- The dataset has the following structure: cluster_<#>/slicing_<#>/schedualing_<#>
+    - 3 clusters (cluster_1, cluster_2, cluster_3)
```
Comment (leonardobonati) [inline README.md]:
```suggestion
    - 3 cluster configurations (cluster_1, cluster_2, cluster_3)
```

File: README.md [hunk 24]
```diff
@@ -3,3 +3,99 @@
...
+    - 5 slicings (slicing_1, slicing_2, slicing_3, slicing_4, slicing_5)
+    - 2 schedualings (scheduling_0, scheduling_2)
+- In total there are 30 combinations of configurations in the available experiments.
+- Here is a tree example of directory structure for `RESERVATION-142634` in cluster_1, slicing_1, schedualing_0 for both datasets.
```
Comment (leonardobonati) [inline README.md]:
```suggestion
- An example of a directory tree example for `RESERVATION-142634` in cluster_1, slicing_1, scheduling_0 for both dataset files.
```

File: README.md [hunk 25]
```diff
@@ -3,3 +3,99 @@
...
+
+# Experimental setup
+- Number of Base Stations (BSs): 1
+- Channel bandwidth: 10 MHz (50 Physical Resource Blocks (PRBs))
+- Number of slices for each BS: 2
+- Scheduling policies available to each slice:
+    - Policy 0: Round-robin (RR)
+    - Policy 2: Proportionally fair (PF)
+- Number of User Equipments (UEs): 8
+    - Users 1-4 demand Enhanced Mobile Broadband (eMBB) traffic
```
Comment (leonardobonati) [inline README.md]:
```suggestion
    - Users 1-4 belong to the enhanced Mobile Broadband (eMBB) slice
```

File: README.md [hunk 26]
```diff
@@ -3,3 +3,99 @@
...
+    - Users 5-8 demand Ultra Reliable and Low Latency Communications (URLLC) traffic
```
Comment (leonardobonati) [inline README.md]:
```suggestion
    - Users 5-8 belong to the Ultra Reliable and Low Latency Communications (URLLC) slice
```

File: README.md [hunk 27]
```diff
@@ -3,3 +3,99 @@
...
+| 1 | 1010123456002 | eMBB | 
+| 2 | 1010123456003 | eMBB |
+| 3 | 1010123456004 | eMBB |
+| 4 | 1010123456005 | eMBB | 
+| 5 | 1010123456006 | URLLC |
+| 6 | 1010123456007 | URLLC |
+| 7 | 1010123456008| URLLC | 
+| 8 | 1010123456009 | URLLC | 
+
+- Traffic: The traffic is twinned from a public dataset of LTE network from 3 BSs located in Madrid, Spain.
```
Comment (leonardobonati) [inline README.md]:
```suggestion
- Traffic: The traffic is twinned from a public dataset of LTE network from 3 BSs located in Madrid, Spain, available at this [link](https://git2.networks.imdea.org/wng/madrid-lte-dataset).
```

File: README.md [hunk 28]
```diff
@@ -3,3 +3,99 @@
 Dataset for the paper L. Bonati, R. Shirkhani, C. Fiandrino, S. Maxenti, S. D'Oro, M. Polese, T. Melodia, "Twinning Commercial Network Traces on Experimental Open RAN Platforms," in Proceedings of ACM WiNTECH, Washington, DC, USA, November 2024.
```
Comment (leonardobonati) [inline README.md]:
```suggestion
Dataset for the paper L. Bonati, R. Shirkhani, C. Fiandrino, S. Maxenti, S. D'Oro, M. Polese, T. Melodia, "Twinning Commercial Network Traces on Experimental Open RAN Platforms," in Proceedings of ACM WiNTECH, Washington, DC, USA, November 2024. [[pdf]](https://arxiv.org/pdf/2409.16217) [[bibtex]](https://ece.northeastern.edu/wineslab/wines_bibtex/bonati2024twinning.txt)
```

**Review by leonardobonati — APPROVED** [review]

**Pull request merged**